### PR TITLE
[chore] Apply clang-format over the whole codebase

### PR DESCRIPTION
**Pull request by zach2good**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I've _**tested my code and the things my code has changed**_ since the last commit in the PR, and will test after any later commits

## What does this pull request do?

I must not have been iterating correctly. Now everything is formatted:
```
for f in $(find src/ -name '*.h' -or -name '*.cpp'); do clang-format-14 -style=file -i $f; done
```

### CODE REVIEW

File: src/map/ai/controllers/automaton_controller.cpp [hunk 1]
```diff
@@ -1642,8 +1661,7 @@ namespace automaton
             {
                 SpellID id = (SpellID)sql->GetUIntData(0);
 
-                AutomatonSpell PSpell
-                {
+                AutomatonSpell PSpell{
```
Comment (claywar) [inline src/map/ai/controllers/automaton_controller.cpp]: Is this intended here, or needs to be another clang off block?

File: src/map/ai/controllers/trust_controller.cpp [hunk 2]
```diff
@@ -282,8 +282,7 @@ void CTrustController::Declump(CCharEntity* PMaster, CBattleEntity* PTarget)
             auto diffAngle  = worldAngle(POwner->loc.p, PTarget->loc.p) + 64;
             auto moveAmount = xirand::GetRandomNumber(0.0f, 1.5f) * ((currentPartyPos % 2) ? 1.0f : -1.0f);
 
-            position_t newPos =
-            {
+            position_t newPos = {
```
Comment (claywar) [inline src/map/ai/controllers/trust_controller.cpp]: Same here

File: src/map/ai/helpers/gambits_container.cpp [hunk 3]
```diff
@@ -433,15 +474,14 @@ namespace gambits
...
+                    auto      mLevel   = POwner->GetMLevel();
 
                     if (action.select == G_SELECT::HIGHEST_WALTZ)
                     {
                         // bool canWaltz = false;
                         auto currentTP = POwner->health.tp;
 
-                        ABILITY wlist[5] =
-                        {
+                        ABILITY wlist[5] = {
```
Comment (claywar) [inline src/map/ai/helpers/gambits_container.cpp]: Same here

File: src/map/lua/lua_zone.cpp [hunk 4]
```diff
@@ -189,7 +189,7 @@ void CLuaZone::reloadNavmesh()
 
 bool CLuaZone::isNavigablePoint(const sol::table& point)
 {
-    position_t position {
+    position_t position{
```
Comment (claywar) [inline src/map/lua/lua_zone.cpp]: Same here

**Review by claywar — APPROVED** [review]

File: src/map/ai/controllers/automaton_controller.cpp (hunk 1 above)
Comment (zach2good) [reply src/map/ai/controllers/automaton_controller.cpp]: Well, these are another thing all together:
This is equivalent to:
```
Type_t t = {};
```
```
Type_t t =
{
    memberInit1,
    memberInit2,
    memberInit3,
};
```

Wasn't sure what to do with these.

Should probably be allman like everything else

**Pull request merged**